src/main/java/dinkplugin/notifiers/KillCountNotifier.java

```diff
@@ -33,27 +41,47 @@ protected String getWebhookUrl() {
         return config.killCountWebhook();
     }
 
+    public void reset() {
+        this.data = null;
+    }
+
     public void onGameMessage(String message) {
         if (isEnabled())
-            parse(message).ifPresent(pair -> handleKill(pair.getLeft(), pair.getRight(), message));
+            parse(message).ifPresent(this::updateData);
+    }
+
+    public void onFriendsChatNotification(String message) {
+        // For CoX, Jagex sends duration via FRIENDSCHATNOTIFICATION
+        if (message.startsWith("Congratulations - your raid is complete!"))
+            this.onGameMessage(message);
     }
 
-    private void handleKill(String boss, int killCount, String rawMessage) {
-        if (!checkKillInterval(killCount)) return;
+    public void onTick() {
+        if (data != null) {
+            // all data must be sent on the same tick to be included
+            handleKill(data);
+            reset();
+        }
+    }
+
+    private void handleKill(BossNotificationData data) {
+        if (data.getBoss() == null || data.getCount() == null || !checkKillInterval(data.getCount(), data.isPersonalBest()))
+            return;
 
         // Assemble content
         String player = Utils.getPlayerName(client);
+        String time = Utils.format(data.getTime(), Utils.isPreciseTiming(client));
         String content = StringUtils.replaceEach(
-            config.killCountMessage(),
-            new String[] { "%USERNAME%", "%BOSS%", "%COUNT%" },
-            new String[] { player, boss, String.valueOf(killCount) }
+            data.isPersonalBest() == Boolean.TRUE ? config.killCountBestTimeMessage() : config.killCountMessage(),
```

Comment:
Can we expand the pb-check to its own variable `isPb` so we can simplify this ternary to just `isPb ? config.. : config...`